src/Compilers/CSharp/Test/Semantic/Semantics/InterceptorsTests.cs

```diff
@@ -1497,6 +1756,470 @@ static class D
+            //     [InterceptsLocation("Program.cs", 15, 11)] // 1
+            Diagnostic(ErrorCode.ERR_InterceptorScopedMismatch, @"InterceptsLocation(""Program.cs"", 15, 11)").WithArguments("C.InterceptableMethod(out int)", "D.Interceptor1(out int)").WithLocation(21, 6)
+            );
+    }
+
+    [Fact]
+    public void ScopedMismatch_04()
+    {
+        // unsafe '[UnscopedRef]' difference
+        var source = """
+            using System.Diagnostics.CodeAnalysis;
+            using System.Runtime.CompilerServices;
+            using System;
+
+            class C
+            {
+                [Interceptable]
+                public static ref int InterceptableMethod(out int value) => throw null!;
+            }
+
+            static class Program
+            {
+                public static void Main()
+                {
+                    C.InterceptableMethod(out int i);
+                }
+            }
+
+            static class D
+            {
+                [InterceptsLocation("Program.cs", 15, 11)] // 1
+                public static ref int Interceptor1([UnscopedRef] out int value) => throw null!;
+            }
+            """;
+        var comp = CreateCompilation(new[] { (source, "Program.cs"), s_attributesSource, (UnscopedRefAttributeDefinition, "UnscopedRefAttribute.cs") }, options: WithNullableEnable());
+        comp.VerifyEmitDiagnostics(
+            // Program.cs(21,6): error CS27019: Cannot intercept call to 'C.InterceptableMethod(out int)' with 'D.Interceptor1(out int)' because of a difference in 'scoped' modifiers or '[UnscopedRef]' attributes.
+            //     [InterceptsLocation("Program.cs", 15, 11)] // 1
+            Diagnostic(ErrorCode.ERR_InterceptorScopedMismatch, @"InterceptsLocation(""Program.cs"", 15, 11)").WithArguments("C.InterceptableMethod(out int)", "D.Interceptor1(out int)").WithLocation(21, 6)
+            );
+    }
+
+    [Fact]
+    public void ReferenceEquals_01()
+    {
+        // A call to 'object.ReferenceEquals(a, b)' is defined as being equivalent to '(object)a == b'.
+        var source = """
+            using System.Runtime.CompilerServices;
+
+            static class D
+            {
+                [Interceptable]
+                public static bool Interceptable(object? obj1, object? obj2) => throw null!;
+
+                public static void M0(object? obj1, object? obj2)
+                {
+                    if (obj1 == obj2)
+                       throw null!;
+                }
+
+                public static void M1(object? obj1, object? obj2)
+                {
+                    if (Interceptable(obj1, obj2))
+                       throw null!;
+                }
+
+                public static void M2(object? obj1, object? obj2)
+                {
+                    if (Interceptable(obj1, obj2))
+                       throw null!;
+                }
+            }
+
+            namespace System
+            {
+                public class Object
```

Comment:
Is there also the reverse scenario where an interceptor outside of `System.Object` sets up an interception on `Object.ReferenceEquals`?

Reply:
Added test `ReferenceEquals_02`